File: backend/app/schemas/course.py
from pydantic import BaseModel


class LessonSummary(BaseModel):
    id: int
    title: str
    order_index: int

    class Config:
        from_attributes = True


class ChapterDetail(BaseModel):
    id: int
    title: str
    order_index: int
    lessons: list[LessonSummary] = []

    class Config:
        from_attributes = True


class CourseSummary(BaseModel):
    id: int
    title: str
    slug: str
    description: str

    class Config:
        from_attributes = True


class CourseDetail(BaseModel):
    id: int
    title: str
    slug: str
    description: str
    chapters: list[ChapterDetail] = []

    class Config:
        from_attributes = True


class LessonDetail(BaseModel):
    id: int
    title: str
    content_md: str
    order_index: int
    chapter_id: int

    class Config:
        from_attributes = True


class QuizQuestionOut(BaseModel):
    id: int
    question_text: str
    options: str  # JSON string
    explanation: str = ""

    class Config:
        from_attributes = True


class QuizOut(BaseModel):
    id: int
    lesson_id: int
    title: str
    questions: list[QuizQuestionOut] = []

    class Config:
        from_attributes = True

File: src-tauri/src/models/course.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct CourseSummary {
    pub id: i64,
    pub title: String,
    pub slug: String,
    pub description: String,
    pub difficulty: String,
    pub duration_minutes: i64,
    pub tags: Vec<String>,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct LessonSummary {
    pub id: i64,
    pub title: String,
    pub order_index: i64,
    pub duration_minutes: i64,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct ChapterDetail {
    pub id: i64,
    pub title: String,
    pub order_index: i64,
    pub lessons: Vec<LessonSummary>,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct CourseDetail {
    pub id: i64,
    pub title: String,
    pub slug: String,
    pub description: String,
    pub difficulty: String,
    pub duration_minutes: i64,
    pub tags: Vec<String>,
    pub chapters: Vec<ChapterDetail>,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct LessonDetail {
    pub id: i64,
    pub title: String,
    pub content_md: String,
    pub order_index: i64,
    pub chapter_id: i64,
    pub duration_minutes: i64,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct QuizQuestionOut {
    pub id: i64,
    pub question_text: String,
    pub options: String,
    pub explanation: String,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct QuizOut {
    pub id: i64,
    pub lesson_id: i64,
    pub title: String,
    pub questions: Vec<QuizQuestionOut>,
}
